music21/expressions.py

```diff
@@ -835,27 +1093,31 @@ class HalfStepMordent(Mordent):
     >>> m = expressions.HalfStepMordent()
     >>> m.direction
     'down'
-    >>> m.size
+    >>> m.getSize(note.Note('C4'))
     <music21.interval.Interval m-2>
     '''
     def __init__(self, **keywords) -> None:
         # no accidental supported here, just "HalfStep"
-        kw = keywords.copy()
-        if 'accid' in kw:
-            del kw['accid']
-        super().__init__(**kw)
-        self._size: interval.IntervalBase = interval.Interval('m-2')
+        if 'accidental' in keywords:
+            raise ExpressionException('Cannot initialize HalfStepMordent with accidental')
+        super().__init__(**keywords)
+        self._minorSecondDown: interval.IntervalBase = interval.Interval('m-2')
```

Comment:
I need to implement `FrozenInterval` to make it so we can cache these things and prevent them from being modified elsewhere.  If it existed then these intervals could be cached.

Reply:
Ok, I'll leave that to you.